osquery/main/main.cpp

```diff
@@ -172,8 +172,11 @@ int startOsquery(int argc, char* argv[], std::function<void()> shutdown) {
   // In either case the watcher may start optionally loaded extensions.
   runner.initWorkerWatcher(kWatcherWorkerName);
 
-  // Begin adhoc io service thread.
-  startIOService();
+  // In profile mode dont start io service thread
+  if (FLAGS_profile == 0) {
+    // Begin adhoc io service thread.
+    startIOService();
```

Comment:
Does this break profiling for the `ec2_instance_metadata` table?

Reply:
Sorry for the confusion! There are some tables that require the IO service to be running. `ec2_instance_metadata` makes an HTTP request.

Reply:
Sorry for late reply.
At present IOService is used for session reuse functionality and session reuse is being used by tls(transport) only(at present).
Making http request is one aspect and keeping the connection alive different. In profile mode table (ec2_instance_metadata)  being  sql'ed and it is using http client, no ioservice is required.